pkg/workloads/containerd/watcher.go

```diff
@@ -139,6 +173,24 @@ func listenForDockerEvents(reader io.ReadCloser) {
 	}
 }
 
+func processContainerEvents(containerID string, events chan dTypesEvents.Message) {
+	for {
```

Comment:
How do we stop these go subroutines?

Reply:
They exit on channel close, which is done by the reapEmpty call. I was trying to avoid more coordination with them but we can have the termination go the other way.